components/event-publisher-proxy/pkg/subscribed/helpers_test.go

```diff
@@ -1,13 +1,154 @@
 package subscribed
 
 import (
+	eventingv1alpha2 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha2"
 	"reflect"
 	"testing"
 
 	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
 )
 
 func TestFilterEventTypeVersions(t *testing.T) {
+	testCases := []struct {
+		name           string
+		appName        string
+		subscription   *eventingv1alpha2.Subscription
+		expectedEvents []Event
+	}{
+		{
+			name:           "should return no events when there is no subscription",
+			appName:        "fooapp",
+			subscription:   &eventingv1alpha2.Subscription{},
+			expectedEvents: make([]Event, 0),
+		}, {
+			name:    "should return a slice of events when eventTypes are provided",
+			appName: "foovarkes",
+			subscription: &eventingv1alpha2.Subscription{
+				Spec: eventingv1alpha2.SubscriptionSpec{
+					Source: "foovarkes",
+					Types: []string{
+						"order.created.v1",
+						"order.created.v2",
+					},
+				},
+			},
+			expectedEvents: []Event{
+				NewEvent("order.created", "v1"),
+				NewEvent("order.created", "v2"),
+			},
+		}, {
+			name:    "should return no event if app name is different than subscription source",
+			appName: "foovarkes",
+			subscription: &eventingv1alpha2.Subscription{
+				Spec: eventingv1alpha2.SubscriptionSpec{
+					Source: "diff-source",
+					Types: []string{
+						"order.created.v1",
+						"order.created.v2",
+					},
+				},
+			},
+			expectedEvents: []Event{},
+		}, {
+			name:    "should return event types w/o prefix for exact external event types",
+			appName: "foovarkes",
+			subscription: &eventingv1alpha2.Subscription{
+				Spec: eventingv1alpha2.SubscriptionSpec{
+					Source:       "/default/sap.kyma/tunas-develop",
+					TypeMatching: eventingv1alpha2.TypeMatchingExact,
+					Types: []string{
+						"sap.kyma.custom.foovarkes.order.created.v1",
+						"sap.kyma.custom.foovarkes.order.created.v2",
+					},
+				},
+			},
+			expectedEvents: []Event{
+				NewEvent("order.created", "v1"),
+				NewEvent("order.created", "v2"),
+			},
+		}, {
+			name:    "should return no event if app name is not part of external event types",
+			appName: "foovarkes",
+			subscription: &eventingv1alpha2.Subscription{
+				Spec: eventingv1alpha2.SubscriptionSpec{
+					Source:       "/default/sap.kyma/tunas-develop",
+					TypeMatching: eventingv1alpha2.TypeMatchingExact,
+					Types: []string{
+						"sap.kyma.custom.difffoovarkes.order.created.v1",
+						"sap.kyma.custom.difffoovarkes.order.created.v2",
+					},
+				},
+			},
+			expectedEvents: []Event{},
+		}, {
+			name:    "should return event type only with 'sap.kyma.custom' prefix",
```

Comment:
can you update this to:
```suggestion
			name:    "should return event type only with 'sap.kyma.custom' prefix and appname",
```